import { defineStore } from 'pinia'
import { createInvoice, getAllInvoices, getInvoiceById } from '../service/api'

export const useInvoiceStore = defineStore('invoice', {
  state: () => ({
    loading: false,
    invoices: [],
    selectedInvoice: null,
    error: null,
  }),
  actions: {
    async fetchInvoices() {
      this.loading = true
      try {
        const res = await getAllInvoices()
        console.log('üìÑ API Response:', res) // ‚úÖ Voir la structure exacte

        // ‚úÖ Adapter selon la structure de l'API

        this.invoices = res.data
        console.warn('‚ö†Ô∏è Structure de donn√©es inattendue:', res.factures)

        this.error = null
        console.log('üìä Invoices loaded:', this.invoices, 'invoices')
        return this.invoices
      } catch (err) {
        this.error = err.message || 'Erreur lors du chargement des factures'
        console.error('‚ùå Error fetching invoices:', err)
        this.invoices = [] // ‚úÖ S'assurer que c'est toujours un tableau
      } finally {
        this.loading = false
      }
    },

    async fetchInvoiceById(id) {
      this.loading = true
      try {
        this.selectedInvoice = await getInvoiceById(id)
        this.error = null
        return this.selectedInvoice
      } catch (err) {
        this.error = err.message || 'Erreur lors du chargement de la facture'
        console.error('‚ùå Error fetching invoice by id:', err)
      } finally {
        this.loading = false
      }
    },

    async createInvoice(invoiceData) {
      console.log('üìù Info received in the store:', invoiceData)
      this.loading = true
      this.error = null
      try {
        const newInvoice = await createInvoice(invoiceData)
        this.invoices.push(newInvoice)
        return newInvoice
      } catch (error) {
        this.error = error.message || 'Erreur lors de la cr√©ation de la facture'
        console.error('‚ùå Erreur lors de la cr√©ation de la facture:', error)
      } finally {
        this.loading = false
      }
    },
  },
})
